src/psbt.h

```diff
@@ -294,6 +385,48 @@ struct PSBTOutput
         // Write any hd keypaths
         SerializeHDKeypaths(s, hd_keypaths, PSBT_OUT_BIP32_DERIVATION);
 
+        // Write the Confidential Assets blinding data
+        if (!value_commitment.IsNull()) {
+            SerializeToVector(s, PSBT_OUT_VALUE_COMMITMENT);
+            SerializeToVector(s, value_commitment);
+        }
+
+        if (!value_blinding_factor.IsNull()) {
+            SerializeToVector(s, PSBT_OUT_VALUE_BLINDER);
+            SerializeToVector(s, value_blinding_factor);
+        }
+
+        // XXX: If g_con_elementsmode is false, something strange and unexpected happens here.
```

Comment:
???

Reply:
Oh, I think all this stuff needs to be gated on g_con_elementsmode.

The specific unexpected thing that happens is that asset_commitment is NOT null; instead it is some kind of special Bitcoin value, so the asset gets printed even when elementsmode is off.

Reply:
Gated all this behind g_con_elementsmode.